engine/gamesys/src/gamesys/scripts/script_physics.cpp

```diff
@@ -1262,6 +1262,198 @@ namespace dmGameSystem
+        lua_setfield(L, -2, "type");
+
+        switch(shape_info.m_Type)
+        {
+            case dmPhysicsDDF::CollisionShape::TYPE_SPHERE:
+                lua_pushnumber(L, shape_info.m_SphereDiameter);
+                lua_setfield(L, -2, "diameter");
+                break;
+            case dmPhysicsDDF::CollisionShape::TYPE_BOX:
+                dmScript::PushVector3(L, dmVMath::Vector3(shape_info.m_BoxDimensions[0], shape_info.m_BoxDimensions[1], shape_info.m_BoxDimensions[2]));
+                lua_setfield(L, -2, "dimensions");
+                break;
+            case dmPhysicsDDF::CollisionShape::TYPE_CAPSULE:
+                lua_pushnumber(L, shape_info.m_CapsuleDiameterHeight[0]);
+                lua_setfield(L, -2, "diameter");
+                lua_pushnumber(L, shape_info.m_CapsuleDiameterHeight[1]);
+                lua_setfield(L, -2, "height");
+            break;
+        }
+
+        return 1;
+    }
+
+    /*# set collision shape data
+     * Sets collision shape data for a collision object. Please note that updating data in 3D
+     * can be quite costly for box and capsules. Because of the physics engine, the cost
+     * comes from having to recreate the shape objects when certain shapes needs to be updated.
+     *
+     * @name physics.set_shape
+     * @param url [type:string|hash|url] the collision object.
+     * @param shape [type:string] the name of the shape to get data for.
+     * @param table [type:table] the shape data to update the shape with.
+     *
+     * See [ref:physics.get_shape] for a detailed description of each field in the data table.
+     *
+     * ```lua
+     * local function set_shape_data()
+     *     -- set capsule shape data
+     *     local data = {}
+     *     data.diameter = 10
+     *     data.height = 20
+     *     physics.set_shape("#collisionobject", "my_capsule_shape", data)
+     *
+     *     -- set sphere shape data
+     *     data = {}
+     *     data.diameter = 10
+     *     physics.set_shape("#collisionobject", "my_sphere_shape", data)
+     *
+     *     -- set box shape data
+     *     data = {}
+     *     data.dimensions = vmath.vector3(10, 10, 5)
+     *     physics.set_shape("#collisionobject", "my_box_shape", data)
+     * end
+     * ```
+     */
+    static int Physics_SetShape(lua_State* L)
+    {
+        DM_LUA_STACK_CHECK(L, 0);
+
+        dmhash_t shape_name_hash = dmScript::CheckHashOrString(L, 2);
+        uint32_t shape_ix;
+        dmGameObject::HCollection collection = dmGameObject::GetCollection(CheckGoInstance(L));
+
+        void* comp       = 0x0;
+        void* comp_world = 0x0;
+        GetCollisionObject(L, 1, collection, &comp, &comp_world);
+
+        dmGameSystem::ShapeInfo shape_info = {};
+
+        if (!dmGameSystem::GetShapeIndex(comp, shape_name_hash, &shape_ix))
+        {
+            return luaL_error(L, "No shape with name '%s' found", dmHashReverseSafe64(shape_name_hash));
```

Comment:
When using DM_LUA_STACK_CHECK, you should use `return DM_LUA_ERROR("No shape with name '%s' found", dmHashReverseSafe64(shape_name_hash))`